src/logic/nontrivial.lean

```diff
@@ -128,3 +133,79 @@ begin
   { exact ⟨x₁, (hf.ne_iff' h).2 hx⟩ },
   { exact ⟨x₂, h⟩ }
 end
+
+namespace tactic
+
+/--
+Given a goal `a = b` or `a ≤ b` in a type `α`, generates an additional hypothesis `nontrivial α`
+(as otherwise `α` is a subsingleton and the goal is trivial).
+-/
+meta def nontriviality_by_elim : tactic unit :=
+do
+  t ←
+    (do t ← mk_mvar, e ← to_expr ``(@eq %%t _ _), target >>= unify e, return t) <|>
+    (do t ← mk_mvar, e ← to_expr ``(@has_le.le %%t _ _ _), target >>= unify e, return t) <|>
```

Comment:
```suggestion
    (do t ← mk_mvar, e ← to_expr ``(@eq %%t _ _), target >>= unify e, instantiate_mvars t) <|>
    (do t ← mk_mvar, e ← to_expr ``(@has_le.le %%t _ _ _), target >>= unify e, instantiate_mvars t) <|>
```

This works without instantiating? I guess it's done in `to_expr` when `t` is used below. (You don't have to make this change since it works, I'm just surprised)